test/unit/app/services/asset_manager/attachment_updater_test.rb

```diff
@@ -4,281 +4,234 @@ class AssetManager::AttachmentUpdaterTest < ActiveSupport::TestCase
+        }
 
-          updater.call(attachment_data, access_limited: true)
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
         end
-      end
 
-      context "when attachment's attachable is not access limited" do
-        before do
-          attachment_data.stubs(:access_limited?).returns(false)
-        end
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, draft_status: true)
+      end
 
-        it "updates all assets to have an empty access_limited array" do
-          update_service.expects(:call)
-                          .with(original_asset.asset_manager_id, attachment_data, nil, { "access_limited_organisation_ids" => [] })
-          update_service.expects(:call)
-                        .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "access_limited_organisation_ids" => [] })
+      it "sets parent_document_url for attachment using draft hostname" do
+        expected_attribute_hash = { "parent_document_url" => edition.public_url(draft: true) }
 
-          updater.call(attachment_data, access_limited: true)
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
         end
-      end
-    end
-
-    describe "draft status updates" do
-      let(:draft) { true }
-      let(:unpublished) { false }
-      let(:replaced) { false }
 
-      before do
-        attachment_data.stubs(:draft?).returns(draft)
-        attachment_data.stubs(:unpublished?).returns(unpublished)
-        attachment_data.stubs(:replaced?).returns(replaced)
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, link_header: true)
       end
 
-      it "marks corresponding assets as draft" do
-        update_service.expects(:call)
-                        .with(original_asset.asset_manager_id, attachment_data, nil, { "draft" => true })
-        update_service.expects(:call)
-                      .with(thumbnail_asset.asset_manager_id, attachment_data, nil, { "draft" => true })
+      it "updates all assets to have an empty access_limited array when the draft is not access limited" do
+        expected_attribute_hash = {
+          "access_limited_organisation_ids" => [],
+        }
+
+        assets.each do |asset|
+          AssetManager::AssetUpdater.expects(:call).with(asset.asset_manager_id, attachment.attachment_data, nil, expected_attribute_hash)
+        end
 
-        updater.call(attachment_data, draft_status: true)
+        AssetManager::AttachmentUpdater.call(attachment.attachment_data, access_limited: true)
       end
 
-      context "and attachment is not draft" do
-        let(:draft) { false }
+      it "marks corresponding assets as not draft even though the attachable is a draft when the attachment has been replaced" do
```

Comment:
I think this can change to `it "marks corresponding assets as not draft when the attachment has been replaced"` to make it more readable. It's already in the context of `"when the attachment's attachable is a draft"` which covers `"even though the attachable is a draft"` part of the test name.